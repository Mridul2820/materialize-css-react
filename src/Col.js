import React from 'react';
import PropTypes from 'prop-types';
import cx from 'classnames';

import { sizesClass } from './tools/sizes';

const extendClassname = (base, classes, namespace) => {
  return namespace.split(' ').reduce((acc, current) => {
    acc[`${base}-${current}`] = true;

    return acc;
  }, classes);
};

const Col = ({
  children,
  className,
  node: C = 'div',
  s,
  m,
  l,
  xl,
  offset,
  push,
  pull,
  ...other
}) => {
  let sizes = { s, m, l, xl };
  let classes = sizesClass({ col: true }, sizes);

  if (typeof offset !== 'undefined') {
    classes = extendClassname('offset', classes, offset);
  }

  if (typeof push !== 'undefined') {
    classes = extendClassname('push', classes, push);
  }

  if (typeof pull !== 'undefined') {
    classes = extendClassname('pull', classes, pull);
  }

  return (
    <C {...other} className={cx(classes, className)}>
      {children}
    </C>
  );
};

Col.propTypes = {
  children: PropTypes.node,
  className: PropTypes.string,
  /**
   * Columns for extra large size screens (Large Desktop Devices > 1200px)
   */
  xl: PropTypes.number,
  /**
   * Columns for large size screens (Desktop Devices > 992px)
   */
  l: PropTypes.number,
  /**
   * Columns for middle size screens (Tablet Devices > 600px)
   */
  m: PropTypes.number,
  /**
   * Columns for small size screens (Mobile Devices <= 600px)
   */
  s: PropTypes.number,
  /**
   * The node to be used for the column
   * @default div
   */
  node: PropTypes.node,
  /**
   * To offset/push/pull, simply add s2 to the class where s signifies the screen
   * class-prefix (s = mobile devices, m = tablet devices, l = desktop devices, xl= large desktop devices) and the number after
   * is the number of columns you want to offset/push/pull by.
   */
  offset: PropTypes.string,
  push: PropTypes.string,
  pull: PropTypes.string
};

export default Col;
