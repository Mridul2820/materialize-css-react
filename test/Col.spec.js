import React from 'react';
import { render } from '@testing-library/react';
import Col from '../src/Col';

describe('<Col />', () => {
  test('renders', () => {
    const { container } = render(
      <Col>
        <p>hello</p>
      </Col>
    );
    expect(container).toMatchSnapshot();
  });

  test('accepts sizes as props', () => {
    const { container } = render(<Col s={4} m={6} l={8} />);
    expect(container).toMatchSnapshot();
  });

  describe('offset', () => {
    test('renders one offset class', () => {
      const { container } = render(<Col offset="s4" />);
      expect(container).toMatchSnapshot();
    });

    test('renders many offset classes', () => {
      const { container } = render(<Col offset="s4 m6 xl8" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('push', () => {
    test('renders one push class', () => {
      const { container } = render(<Col push="s4" />);
      expect(container).toMatchSnapshot();
    });

    test('renders many push classes', () => {
      const { container } = render(<Col push="s4 m6 xl8" />);
      expect(container).toMatchSnapshot();
    });
  });

  describe('pull', () => {
    test('renders one pull class', () => {
      const { container } = render(<Col pull="s4" />);
      expect(container).toMatchSnapshot();
    });

    test('renders many pull classes', () => {
      const { container } = render(<Col pull="s4 m6 xl8" />);
      expect(container).toMatchSnapshot();
    });
  });
});
