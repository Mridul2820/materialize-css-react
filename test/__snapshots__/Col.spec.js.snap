// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Col /> accepts sizes as props 1`] = `
<div>
  <div
    class="col s4 m6 l8"
  />
</div>
`;

exports[`<Col /> offset renders many offset classes 1`] = `
<div>
  <div
    class="col offset-s4 offset-m6 offset-xl8"
  />
</div>
`;

exports[`<Col /> offset renders one offset class 1`] = `
<div>
  <div
    class="col offset-s4"
  />
</div>
`;

exports[`<Col /> pull renders many pull classes 1`] = `
<div>
  <div
    class="col pull-s4 pull-m6 pull-xl8"
  />
</div>
`;

exports[`<Col /> pull renders one pull class 1`] = `
<div>
  <div
    class="col pull-s4"
  />
</div>
`;

exports[`<Col /> push renders many push classes 1`] = `
<div>
  <div
    class="col push-s4 push-m6 push-xl8"
  />
</div>
`;

exports[`<Col /> push renders one push class 1`] = `
<div>
  <div
    class="col push-s4"
  />
</div>
`;

exports[`<Col /> renders 1`] = `
<div>
  <div
    class="col"
  >
    <p>
      hello
    </p>
  </div>
</div>
`;
